main/src/main/java/com/sedmelluq/discord/lavaplayer/player/DefaultAudioPlayer.java

```diff
@@ -241,9 +241,13 @@ public boolean provide(MutableAudioFrame targetFrame, long timeout, TimeUnit uni
     private void handleTerminator(InternalAudioTrack track) {
         synchronized (trackSwitchLock) {
             if (activeTrack == track) {
-                activeTrack = null;
-
-                dispatchEvent(new TrackEndEvent(this, track, track.getActiveExecutor().failedBeforeLoad() ? LOAD_FAILED : FINISHED));
+                InternalAudioTrack oldTrack = activeTrack;
```

Comment:
Actually, I do have one question -- why assign a variable here rather than use the `track` variable?

Reply:
Becuase `oldTrack` is stopped after `activeTrack` is set to null.

Reply:
I think devo meant the `track` variable used in the event. That one seems to not be nulled